packages/web3-eth-abi/src/utils.ts

```diff
@@ -123,12 +123,18 @@ export const mapTypes = (
 	return mappedTypes;
 };
 
+/**
+ * format odd-length bytes to even-length
+ */
+export const formatOddHexstrings = (param: string): string =>
+	param.length % 2 === 1 ? `0x0${param.substring(2)}` : param;
+
 /**
  * Handle some formatting of params for backwards compatibility with Ethers V4
  */
 export const formatParam = (type: string, _param: unknown): unknown => {
 	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-	let param = _param;
+	const param = _param;
```

Comment:
More safe approach is to clone `_param` obj, as `param` is also returned in this function.